src/CoreGraphics/CGPDFDocument.cs

```diff
@@ -258,6 +258,37 @@ public CGPDFDictionary GetInfo ()
 		{
 			return new CGPDFDictionary (CGPDFDocumentGetInfo (handle));
 		}
+
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[iOS (11,0), Mac(10,3), TV(11,0), Watch(4,0)]
+		extern static void CGPDFContextSetOutline (/* CGPDFDocumentRef */ IntPtr document, IntPtr /* dictionary */ outline);
+
+		public void SetOutline (CGPDFOutlineOptions options)
+		{
+			CGPDFContextSetOutline (handle, options == null ? IntPtr.Zero : options.Dictionary.Handle);
+		}
+					
+		[DllImport (Constants.CoreGraphicsLibrary)]
+		[iOS (11,0), Mac(10,3), TV(11,0), Watch(4,0)]
+		extern static /* CFDictionaryPtry */ IntPtr CGPDFDocumentGetOutline (/* CGPDFDocumentRef */ IntPtr document);
+
+		[iOS (11,0), Mac(10,3), TV(11,0), Watch(4,0)]
+		public CGPDFOutlineOptions GetOutline ()
+		{
+			var ptr = CGPDFDocumentGetOutline (handle);
+			return new CGPDFOutlineOptions (new NSDictionary (ptr));
```

Comment:
Use `Runtime.GetNSObject<NSDictionary> (ptr)` instead of `new NSDictionary (ptr)`, since the former will fetch any existing managed NSDictionary instances.